crates/editor/src/editor.rs

```diff
@@ -1994,6 +1997,12 @@ impl Editor {
         }
 
         self.blink_manager.update(cx, BlinkManager::pause_blinking);
+        if ProjectSettings::get_global(cx)
+            .diagnostics
+            .update_with_cursor()
+        {
+            self.show_active_diagnostic_at_cursor(cx);
```

Comment:
It looks like if I navigate into a line with the diagnostics, open that and move one line up/down to a blank space, the diagnostics is not gone:

https://github.com/user-attachments/assets/6dc82cf6-86e6-4d1b-91fe-edab8b5dd253

that's not what Helix does, rather removing the expansions instantly.
I like the latter more, but maybe there was a reason to do it otherwise?

Reply:
Absolutely! This was mostly just due to how I reused the existing active diagnostics group machinery. After using this for a few days full time, the `update_with_cursor` option needs to auto-dismiss when moving out of a diagnostic. Needing to reach the escape key to dismiss something you looked at has been fairly annoying.